wasm/jit/jit_amd64_test.go

```diff
@@ -1350,11 +1370,28 @@ func TestAmd64Compiler_emitEqOrNe(t *testing.T) {
 					{x1: 6.8719476736e+10 /* = 1 << 36 */, x2: 1.37438953472e+11 /* = 1 << 37*/},
 					{x1: math.Inf(1), x2: 100},
 					{x1: math.Inf(-1), x2: 100},
+					{x1: math.NaN(), x2: 1.231},
+					{x1: math.NaN(), x2: -1.231},
+					{x1: 1.231, x2: math.NaN()},
+					{x1: -1.231, x2: math.NaN()},
+					{x1: math.Inf(1), x2: 100},
+					{x1: 100, x2: math.Inf(1)},
+					{x1: math.Inf(1), x2: math.Inf(1)},
+					{x1: math.Inf(-1), x2: 100},
+					{x1: 100, x2: math.Inf(-1)},
+					{x1: math.Inf(-1), x2: math.Inf(-1)},
 				} {
 					t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
 						compiler := requireNewCompiler(t)
 						compiler.initializeReservedRegisters()
 
+						// For float operations, we have to reserve two int registers to deal with NaN cases.
```

Comment:
extract to a function to reduce copy/paste drift if possible... I know bit length might not make that work, so just think about it and resolve this comment either way